src/components/AccountDrawer/PrefetchBalancesWrapper.tsx

```diff
@@ -3,36 +3,19 @@ import { usePortfolioBalancesLazyQuery, usePortfolioBalancesQuery } from 'graphq
 import { GQL_MAINNET_CHAINS } from 'graphql/data/util'
 import usePrevious from 'hooks/usePrevious'
 import { atom, useAtom } from 'jotai'
-import { PropsWithChildren, useCallback, useEffect, useMemo } from 'react'
-import { useAllTransactions } from 'state/transactions/hooks'
-import { TransactionDetails } from 'state/transactions/types'
+import { PropsWithChildren, useCallback, useEffect } from 'react'
+import { usePendingOrders } from 'state/signatures/hooks'
+import { usePendingTransactions } from 'state/transactions/hooks'
 
-const isTxPending = (tx: TransactionDetails) => !tx.receipt
-function wasPending(previousTxs: { [hash: string]: TransactionDetails | undefined }, current: TransactionDetails) {
-  const previousTx = previousTxs[current.hash]
-  return previousTx && isTxPending(previousTx)
-}
-
-function useHasUpdatedTx(account: string | undefined) {
-  // TODO: consider monitoring tx's on chains other than the wallet's current chain
-  const currentChainTxs = useAllTransactions()
-
-  const pendingTxs = useMemo(() => {
-    return Object.entries(currentChainTxs).reduce((acc: { [hash: string]: TransactionDetails }, [hash, tx]) => {
-      if (!tx.receipt) acc[hash] = tx
-      return acc
-    }, {})
-  }, [currentChainTxs])
+/** Returns true if the number of pending activities has decreased */
+function useHasUpdatedTx() {
+  const numPendingTxs = usePendingTransactions().length
+  const numPendingOrders = usePendingOrders().length
 
-  const previousPendingTxs = usePrevious(pendingTxs)
+  const totalPending = numPendingTxs + numPendingOrders
+  const prevPending = usePrevious(totalPending)
 
-  return useMemo(() => {
-    if (!previousPendingTxs || !account) return false
-    return Object.values(currentChainTxs).some(
-      (tx) => tx.from === account && !isTxPending(tx) && wasPending(previousPendingTxs, tx),
-      [currentChainTxs, previousPendingTxs]
-    )
-  }, [account, currentChainTxs, previousPendingTxs])
+  return !!prevPending && totalPending < prevPending
```

Comment:
I realized this hook was very overcomplicated and could be replaced w/ just counting difference in pending activity count between renders

Reply:
nice